File: Client_Start/Assets/LuaFramework/Lua/MyLua/Libs/Table/ItemObject/TableState.lua
MLoader("MyLua.Libs.Core.GlobalNS");
MLoader("MyLua.Libs.Core.Class");

local M = GlobalNS.Class(GlobalNS.TableItemBodyBase);
M.clsName = "TableStateItemBody";
GlobalNS[M.clsName] = M;

function M:ctor()
    self.m_name = "";           -- 名称
    self.m_res = "";            -- 资源
    self.m_effectId = 0;        -- 特效 Id
end

function M:parseBodyByteBuffer(bytes, offset)
    bytes.position = offset;
    self.m_name = GlobalNS.UtilTable.readString(bytes, self.m_name);
    self.m_res = GlobalNS.UtilTable.readString(bytes, self.m_res);
    _, self.m_effectId = bytes:readInt32(self.m_effectId);

    self:initDefaultValue();
end

function M:initDefaultValue()
    if(self.m_effectId == 0) then
        self.m_effectId = 0;
    end
end

return M;

File: PlaneWar/Client_1/Assets/Scripts/Libs/Table/ItemObject/TableState.cs
namespace SDK.Lib
{
    public class TableStateItemBody : TableItemBodyBase
    {
        public string mName;           // 名称
        public string mRes;            // 资源
        public int mEffectId;          // 特效 Id

        override public void parseBodyByteBuffer(ByteBuffer bytes, uint offset)
        {
            bytes.position = offset;
            UtilTable.readString(bytes, ref mName);
            UtilTable.readString(bytes, ref mRes);
            bytes.readInt32(ref mEffectId);

            initDefaultValue();
        }

        protected void initDefaultValue()
        {
            if(mEffectId == 0)
            {
                mEffectId = 0;
            }
        }
    }
}